src/backend/executor/nodeHash.c

```diff
@@ -366,9 +400,23 @@ MultiExecParallelHash(HashState *node)
 				 * skew).
 				 */
 				pstate->growth = PHJ_GROWTH_DISABLED;
+				/* In case we didn't find null values ourself. */
+				if (pstate->phs_lasj_has_null)
+				{
+					node->hs_hashkeys_null = true;
+					return;
+				}
 			}
 	}
 
+	/* In case we didn't participate in PHJ_BUILD_HASHING_INNER */
+	pg_memory_barrier();
```

Comment:
use pg_write_barrier(); ?

Reply:
pg_memory_barrier is more strict, if we write variable by ourself  pg_write_barrier() is better.